include/curl/curl.h

```diff
@@ -1726,6 +1726,7 @@ enum {
   CURL_HTTP_VERSION_1_0,  /* please use HTTP 1.0 in the request */
   CURL_HTTP_VERSION_1_1,  /* please use HTTP 1.1 in the request */
   CURL_HTTP_VERSION_2_0,  /* please use HTTP 2 in the request */
+  CURL_HTTP_VERSION_2_0_CLEAN_TCP,  /* please use HTTP 2 over clean TCP */
```

Comment:
1. "CLEAN_TCP" sounds odd to me, normal Upgrade: HTTP/2 is also "clean TCP". The term "prior knowledge" is used in the spec but "no upgrade" is also better I think.

2. It can't be inserted _before_ CURL_HTTP_VERSION_2TLS since it is an enum and we have a stable API so we can't change numbers for already delivered options.